source/templates/code_signing.slim

```diff
@@ -106,24 +106,28 @@
 						.menu.small
 							select.menu-selector[ng-model="provProfileCtrl.selectedMenu" ng-model-options="{getterSetter: true}" ng-options="menu.title for menu in provProfileCtrl.menus" ng-class="{ 'selected' : menu == provProfileCtrl.selectedMenu }"]
 					.content.vertical-scroll[ng-switch="provProfileCtrl.selectedMenu.id"]
+						.error-notice[ng-if="!provProfileCtrl.hasValidIdentity()" ng-switch-when="matching-certificates"] == data[:strings][:code_signing][:prov_profile][:prov_profile_details_popup][:matching_certificates_details][:missing_or_expired_notice]
 						ul.matching-certificates[ng-switch-when="matching-certificates"]
-							li.identity[ng-repeat="identity in provProfileCtrl.selectedProvProfile.developerCertificateIdentities"]
+							li.identity[ng-repeat="identity in provProfileCtrl.selectedProvProfile.developerCertificateIdentities" ng-class="{'missing': provProfileCtrl.identityStatusCSSclass(identity) == 'missing'}"]
 								.status[ng-class="provProfileCtrl.identityStatusCSSclass(identity)"]
 									.icon-wrapper.loading-icon-wrapper = svg("dot")
-									.icon-wrapper.missing-icon-wrapper = svg("triangle")
-									.icon-wrapper.ok-icon-wrapper = svg("check-in-circle")
+									.icon-wrapper.missing-icon-wrapper = svg("dot")
+									.icon-wrapper.expired-icon-wrapper = svg("warning-in-circle")
+									.icon-wrapper.ok-icon-wrapper = svg("tick-in-circle")
 								.details
 									.column
 										.common-name[ng-bind="identity.commonName"]
 										.serial[ng-bind="identity.serial"]
 									.column
-										.expired[ng-if="provProfileCtrl.isIdentityExpired(identity)"] == data[:strings][:code_signing][:prov_profile][:prov_profile_details_popup][:identities][:expired]
+										.uploaded[ng-if="provProfileCtrl.identityStatusCSSclass(identity) == 'ok'"] == data[:strings][:code_signing][:prov_profile][:prov_profile_details_popup][:identities][:uploaded]
```

Comment:
`.identityStatusCSSclass()` and other functions that return CSS classes should not be used for condition checks like an `ng-if`, only for `ng-class`. Use `isIdentityWithSerialAddedToApp()` instead, you will have to make it a `viewModel.` function.